examples/backends/differentiable_backend.py

```diff
@@ -1,22 +1,33 @@
+#!/bin/python
 from __future__ import annotations
 
+from pathlib import Path
+
 import numpy as np
 import sympy
 import torch
 
-from qadence import (
-    CNOT,
-    RX,
-    RY,
-    Parameter,
-    QuantumCircuit,
-    chain,
-    total_magnetization,
-)
+DEVICE = torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
+torch.set_default_device(DEVICE)
+torch.manual_seed(42)
+if DEVICE == torch.device("cuda"):
+    try:
+        import os
+
+        os.system("pip install nvidia-pyindex")
+        os.system("pip install nvidia-dlprof[pytorch]")
```

Comment:
IDK if we should leave this in, but sure we can leave the dlprof in. Although I wonder if it would be better off in a specific example then, because this has pretty large performance penalty

Reply:
I changed this already. Its now a separate script with those lines commented